### DISCUSSION TITLE
Delete connections using User name

### DISCUSSION THREAD

**Discussion by NuwanSameera** [rabbitmq/rabbitmq-server#5196] [Other] [answered]
I use Rabbit MQ as my message broker. I need to remove existing connections by user name. Current Management API remove connection by connection_name. 
Is there anyway to achieve this use case without modify rabbitmq_management plugin.

UPDATE : I use auth_backend to authenticate user. So user account not available in rabbit-mq user registry

**Comment by michaelklishin** [comment, marked as answer]
See [Revoking User Access](https://rabbitmq.com/access-control.html#revoke).

Another option would be to list connections using `rabbitmqctl list_connections name user`, filter them and close them one by one. But in most cases, deleting the user is the easiest option.

**Reply by NuwanSameera** [reply]
Thank you for your answer. In my application I use auth_backend. User not available in Rabbit_MQ user list. So user revoke is not a possible option. 
Also REST call is the most possible option for my use case

**Reply by michaelklishin** [reply]
You can list connections using the HTTP API, filter them in your app and close a subset of connections. You can also limit the number of connections per user. While lowering it won't close existing connections, this may be close to your end goal.

**Reply by NuwanSameera** [reply]
Is it possible to add delete connection method to management agent plugin ?

**Reply by michaelklishin** [reply]
HTTP API already provides an endpoint for that: `DELETE /api/connections/{connection_name}`. The management agent plugin does not provide the API but `rabbitmq_management` does.

**Reply by NuwanSameera** [reply]
Yes. My requirement is need to delete connection by user_name not connection name. Because connection name cannot generate using known parameters.

**Reply by michaelklishin** [reply]
I don't have much to add to my recommendation above.

**Reply by lukebakken** [reply]
> My requirement is need to delete connection by user_name not connection name. Because connection name cannot generate using known parameters

List the connections and you can get connection name and user name from it. Just like @michaelklishin said here: https://github.com/rabbitmq/rabbitmq-server/discussions/5196#discussioncomment-3123743

You can also use the HTTP API to do this.

**Reply by mkuratczyk** [reply]
There's no need to filter on your own, there's a dedicated command:
rabbitmqctl close_all_user_connections
